gpkit/nomials/nomial_math.py

```diff
@@ -591,97 +468,179 @@ def __init__(self, left, right, oper_ge=True):
                     p = p/coeff
                 elif p.cs[i] > 1:
                     raise ValueError("infeasible constraint:"
-                                     "constant term too large.")
-
-        super(Constraint, self).__init__(p)
-        self.__class__ = Constraint  # TODO should not have to do this
+                                     " constant term too large.")
+        self.p_lt, self.m_gt = p_lt, m_gt
+        self.posylt1_rep = p
+        self.substitutions = p.values
+
+    def as_posyslt1(self):
+        posys = listify(self.posylt1_rep)
+        if not self.substitutions:
+            # just return the pre-generated posynomial representation
+            return posys
+
+        out = []
+        for posy in posys:
+            if hasattr(self, "m_gt"):
+                m_gt = self.m_gt.sub(self.substitutions,
+                                     require_positive=False)
+                if m_gt.c == 0:
+                    return []
+
+            _, exps, cs, _ = substitution(posy, self.substitutions)
+            # remove any cs that are just nans and/or 0s
+            nans = np.isnan(cs)
+            if np.all(nans) or np.all(cs[~nans] == 0):
+                return []  # skip nan'd or 0'd constraint
+
+            exps, cs, pmap = simplify_exps_and_cs(exps, cs, return_map=True)
+
+            #  The monomial sensitivities from the GP/SP are in terms of this
+            #  smaller post-substitution list of monomials, so we need to map
+            #  back to the pre-substitution list.
+            #
+            #  A "pmap" is a list of HashVectors (mmaps), whose keys are
+            #  monomial indexes pre-substitution, and whose values are the
+            #  percentage of the simplified  monomial's coefficient that came
+            #  from that particular parent.
+
+            self.pmap = pmap
```

Comment:
We could, but I think that's a design pattern that users might also be implementing when writing their own constraints (for example Xu)